### DISCUSSION TITLE
Missing the "Loading" Control.

### DISCUSSION THREAD

**Discussion by tinodo** [CommunityToolkit/Windows#280] [Feedback]
While upgrading from a previous version of the controls, I see that the Loading control hasn't made it to v8.
Any plans on adding this back in?

Thanks!

**Comment by tinodo** [comment]
I see there was a PR for this already: https://github.com/CommunityToolkit/Windows/pull/242
Perhaps this can go through?

**Reply by michael-hawker** [reply]
We decided to instead use the more generalized SwitchPresenter already in the toolkit and show how it can be used for this experience instead, that PR for a sample is here: https://github.com/CommunityToolkit/Windows/pull/285

It allows for more customization as well as additional states if you want to more easily show specific content if an error is encountered while loading, for instance.

**Reply by zhuxb711** [reply]
Without Loading, imagine you have a photo application, user want to save their changes on a image. If there only "SwitchPresenter" available, the user would find out that their image dispeared (in order to show the progress), and the image back (changes save to file and we give user the chance to modify the image again)

@michael-hawker Do you think that is a good user experience? If the experience is excellent, please let me know why the "Photos" application installed on Windows also uses a control that similar to "Loading" rather than "SwitchPresenter "

![image](https://github.com/user-attachments/assets/2c05789f-0a7d-40a2-8573-65f91ecd6508)

**Reply by michael-hawker** [reply]
@zhuxb711 applications are built in a variety of ways and have a variety of requirements. There's many ways to create and layer UI experiences. I don't know what the Photos app is doing here, it's probably just a `ProgressRing` in its own `Grid` vs. the toolkit's deprecated `Loading` control. Your scenario can just as easily fall apart with the old Loading control as well, it was meant as a UI placeholder for displaying content instead of something else too and a wrapper around visibility containing a `ProgressRing`.

The `SwitchPresenter` is a generalized control useful in a variety of scenarios. A photos app developer doesn't have to use it only to display their image OR a loading state; they can layer it on top in order to show the `ProgressRing` when work is being performed and nothing when it's done without doing anything with the image.

We have shipped the referenced sample in the update to our gallery app; while that specific scenario shows swapping a `ProgressRing` with content, it's not the only way to use the control or create these experiences. 

We can't maintain and have solutions for every individual scenario, that's infeasible. But we can provide the building blocks and tools which provide solutions for a variety of scenarios, like these and others. That's where the `SwitchPresenter` shines. I encourage you to understand how to leverage its power for these and other scenarios.
